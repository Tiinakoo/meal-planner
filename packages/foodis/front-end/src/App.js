import "./App.scss";
import "./Typography.scss";
import React from "react";
import CreateDepartment from "./departments/CreateDepartment";
import CreateDepartmentModel from "./departments/CreateDepartmentModel";

const createDepartmentModel = new CreateDepartmentModel();

function App() {
  return (
    <div className="App">
      <div className="page">
        <div className="panel">
          <div className="typography__primary">
            <CreateDepartment createDepartmentModel={createDepartmentModel} />
          </div>
        </div>
      </div>
    </div>
  );
}

export default App;
