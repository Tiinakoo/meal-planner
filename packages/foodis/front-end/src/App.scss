@import './styles/colors';
@import './styles/mixins';
@import './styles/baseline';

.App {
  text-align: center;
}

.page {
  width: 100%;
  margin: auto;
}

.panel {
  margin: 5 * $baseline auto;
  background-color: $secondaryColor;
  padding: 6 * $baseline;
  max-width: 60 * $baseline;

  @include md {
    max-width: 80 * $baseline;
  }

  @include lg {
    max-width: 120 * $baseline;
  }
}

.cards {
  margin: 0 auto;
  display: grid;
  grid-gap: 1rem;
  align-items: center;
  grid-template-columns: repeat(auto-fit, minmax(320px, 1fr))
}

 .card {
   background-color: $primaryColor;
   padding: 1rem;
   height: 4rem;
}
