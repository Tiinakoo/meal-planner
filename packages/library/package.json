{
  "name": "@root/shared",
  "version": "0.0.0",
  "description": "shared library",
  "homepage": "",
  "main": "lib/library.js",
  "directories": {
    "lib": "lib"
  },
  "files": [
    "lib"
  ],
  "devDependencies": {
    "@babel/cli": "^7.12.1",
    "@babel/core": "7.2.2",
    "@babel/plugin-proposal-class-properties": "^7.4.4",
    "@babel/plugin-proposal-decorators": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/runtime": "^7.3.4"
  }
}
