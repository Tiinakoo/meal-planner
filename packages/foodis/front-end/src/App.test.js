import App from './App';
import { mount } from 'enzyme';
import React from 'react';
import '../../../library/setupUnitTests'

describe('App', () => {
  let component;

   beforeEach(() => {
     component = mount(<App />);
   });

  it('renders', () => {
    expect(component).toMatchSnapshot();
  });
});

