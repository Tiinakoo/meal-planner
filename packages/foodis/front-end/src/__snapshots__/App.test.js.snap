// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="App"
>
  <div
    className="page"
  >
    <div
      className="panel"
    >
      <div
        className="typography__primary"
      >
        <Memo(wrappedComponent)
          createDepartmentModel={
            CreateDepartmentModel {
              "dependencies": Object {
                "submitDepartment": [Function],
              },
              "name": TextInputModel {
                "required": true,
                "value": "",
              },
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
